packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.ts

```diff
@@ -0,0 +1,462 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import {
+    TreeWidget,
+    ContextMenuRenderer,
+    CompositeTreeNode,
+    ExpandableTreeNode,
+    SelectableTreeNode,
+    TreeModel,
+    TreeNode,
+    NodeProps,
+    LabelProvider,
+    TreeExpansionService,
+    ApplicationShell,
+    DiffUris
+} from "@theia/core/lib/browser";
+import { SearchInWorkspaceResult, SearchInWorkspaceOptions } from "../common/search-in-workspace-interface";
+import { SearchInWorkspaceService } from "./search-in-workspace-service";
+import { TreeProps } from "@theia/core/lib/browser";
+import { EditorManager, EditorDecoration, TrackedRangeStickiness, OverviewRulerLane, EditorWidget, ReplaceOperation } from "@theia/editor/lib/browser";
+import { inject, injectable } from "inversify";
+import URI from "@theia/core/lib/common/uri";
+import { Path, CancellationTokenSource, Emitter, Event } from "@theia/core";
+import { WorkspaceService } from "@theia/workspace/lib/browser";
+import { h } from "@phosphor/virtualdom";
+import { MEMORY_TEXT } from "./memory-text-resource";
+import { FileResourceResolver } from "@theia/filesystem/lib/browser";
+
+export interface ResultNode extends ExpandableTreeNode, SelectableTreeNode {
+    children: ResultLineNode[];
+    path: string;
+    file: string;
+}
+export namespace ResultNode {
+    export function is(node: any): node is ResultNode {
+        return ExpandableTreeNode.is(node) && SelectableTreeNode.is(node) && "path" in node;
+    }
+}
+
+export type ResultLineNode = SelectableTreeNode & SearchInWorkspaceResult;
+export namespace ResultLineNode {
+    export function is(node: any): node is ResultLineNode {
+        return SelectableTreeNode.is(node) && "line" in node && "character" in node && "lineText" in node;
+    }
+}
+
+@injectable()
+export class SearchInWorkspaceResultTreeWidget extends TreeWidget {
+
+    protected resultTree: Map<string, ResultNode>;
+    protected workspaceRoot: string = "";
+
+    protected _showReplaceButtons = false;
+    protected _replaceTerm = "";
+    protected searchTerm = "";
+
+    protected appliedDecorations = new Map<string, string[]>();
+
+    private cancelIndicator = new CancellationTokenSource();
+
+    protected changeEmitter: Emitter<Map<string, ResultNode>>;
+
+    constructor(
```

Comment:
https://github.com/theia-ide/theia/wiki/Coding-Guidelines#property-injection please review for other occurences

Reply:
done